djangocms_pageadmin/admin.py

```diff
@@ -53,15 +55,36 @@ def get_list_display(self, request):
     def get_queryset(self, request):
         """Filter PageContent objects by current site of the request.
         """
+        url_subquery = PageUrl.objects.filter(
+            language=OuterRef("language"), page=OuterRef("page")
+        )
+        draft_version_lock_subquery = VersionLock.objects.filter(
+            version__content_type=OuterRef("content_type"),
+            version__object_id=OuterRef("object_id"),
+            version__state=DRAFT,
+        ).order_by("-pk")
         queryset = (
             super()
             .get_queryset(request)
             .filter(page__node__site=get_current_site(request))
+            .annotate(_path=Subquery(url_subquery.values("path")[:1]))
+        )
+        return queryset.select_related("page").prefetch_related(
+            Prefetch(
+                "versions",
+                queryset=Version.objects.annotate(
+                    # used by locking
+                    _draft_version_user_id=Subquery(
+                        draft_version_lock_subquery.values("created_by")[:1]
+                    )
+                )
+                .select_related("created_by", "versionlock")
+                .prefetch_related("content"),
+            )
         )
-        return queryset.prefetch_related("versions")
 
     def get_version(self, obj):
-        return Version.objects.get_for_content(obj)
+        return obj.versions.all()[0]
```

Comment:
Would it not be better to use .first() here rather than [0]?

Reply:
`.first()` bypasses prefetch_related cache, needs to use `.all()[0]`